x-pack/plugins/fleet/server/services/security/message_signing_service.ts

```diff
@@ -135,26 +133,38 @@ export class MessageSigningService implements MessageSigningServiceInterface {
     return publicKey;
   }
 
-  public async rotateKeyPair(): Promise<boolean> {
-    const isRemoved = await this.removeKeyPair();
-    if (isRemoved) {
+  public async rotateKeyPair(): ReturnType<MessageSigningServiceInterface['rotateKeyPair']> {
+    try {
+      const removeKeyPairResponse = await this.removeKeyPair();
+      if (!removeKeyPairResponse) {
+        return;
+      }
       await this.generateKeyPair();
-      return true;
+    } catch (error) {
+      throw Error(`Error rotating key pair: ${error.message}`);
```

Comment:
I feel like you should be logging `error` and not just pass the `error.message` along. Example: for ES/SO errors, the details found in the error itsself would be helpful in debugging. 

Suggestion: I just took a look at Fleet's error handler for APIs and noticed that they have special processing for `FleetError`'s. My suggestion is that you  create a new Error type for the use by the message signing service here:

https://github.com/elastic/kibana/blob/140f7207105431272c8d60840a8a0cc254b17124/x-pack/plugins/fleet/common/errors.ts#L20-L19

That extends `FleetError` and then throw that instead.

example:

```typescript
export class MessageSigningError extends FleetError {}
```

 `FleetError` seems to support providing it a `meta` property when creating the error, which you can use to store the original error received. and you could also set `attributes` on the Error, which are then returned in the API error response. Ideally you will capture data from the SO service call error. 

Example:

```typescript
throw new MessageSigningError(`Error rotating key pair: ${error.message}`, error);
```

In the API handler, prior to passing the above error to the fleet's default error handler, since `meta` is not logged via the default fleet error handler, you could then log the real error to kibana logs.



```typescript
log.error(error.meta)
```

Hope this make sense.


_(FYI: one way to test this locally and see whats available in the error is to force the SO call to fail - like give it an invalid SO Type, or invalid ID, or invalid payload - and then `console.log` it using  `util.inspect(data, { depth: 10 }))`)_

Reply:
Thanks @paul-tavares for the detailed info. I'll go ahead and log the errors as well.

Reply:
done 266456977a6b10511bdddbc432b8442a3d601730

Reply:
also added some tests for the errors 4a1f8643cc86c825893fca11443e554ce38a04b4